File: linked_lists/single_ll.py
class Node:
    def __init__(self, data) -> None:
        self.data = data
        self.next = None


# Create first node
head = Node(10)

# Link the second node
head.next = Node(20)  # type: ignore

head.next.next = Node(30)  # type: ignore


def traverse(head: Node):
    pointer = head
    while pointer is not None:
        print(pointer.data, end=" ")
        pointer = pointer.next


traverse(head)

File: linked_lists/SingleLL.js
"use strict";

class Node {
  constructor(data) {
    this.data = data;
    this.next = null;
  }
}

const head = new Node(10);
head.next = new Node(20);
head.next.next = new Node(30);

function traverse(head) {
  let pointer = head;
  while (pointer !== null) {
    console.log(`[${pointer.data}]->`);
    pointer = pointer.next;
  }
}

traverse(head);
